### PR TITLE
[Doc] Update ChatQnA flow chart

### PR DESCRIPTION
**Pull request by xiguiw**
Draw flow-chat with mermaid script. 
make it easy to be maintained.

### CODE REVIEW

**Comment by xiguiw** [comment]
The output flow-chart is:
image

**Review by louie-tsai — COMMENTED** [review]
some minor changes.
Might have more feedbacks but we might need to bring up the different solution (langchain, llama-index, and etc) used in microservice.

**Review by hshen14 — APPROVED** [review]

File: ChatQnA/README.md [hunk 1]
```diff
@@ -10,7 +10,69 @@ ChatQnA architecture shows below:
 
 ChatQnA is implemented on top of [GenAIComps](https://github.com/opea-project/GenAIComps), the ChatQnA Flow Chart shows below:
 
-![Flow Chart](./assets/img/chatqna_flow_chart.png)
+```mermaid
+graph LR
+    subgraph ChatQnA Mega Service
```
Comment (louie-tsai) [inline ChatQnA/README.md:15]:
```suggestion
    subgraph ChatQnA-MegaService["ChatQnA-MegaService"]
```

File: ChatQnA/README.md [hunk 2]
```diff
@@ -10,7 +10,69 @@ ChatQnA architecture shows below:
...
+        direction LR
+        EM([OPEA Embedding<br>6000])
```
Comment (louie-tsai) [inline ChatQnA/README.md:17]:
```suggestion
        EM([Embedding 'LangChain TEI' <br>6000])
```

File: ChatQnA/README.md [hunk 3]
```diff
@@ -10,7 +10,69 @@ ChatQnA architecture shows below:
...
+        RET([OPEA Retrieval<br>7000])
```
Comment (louie-tsai) [inline ChatQnA/README.md:18]:
```suggestion
        RET([Retrieval 'LangChain Redis'<br>7000])
```

File: ChatQnA/README.md [hunk 4]
```diff
@@ -10,7 +10,69 @@ ChatQnA architecture shows below:
...
+        RER([OPEA Reranking<br>8000])
```
Comment (louie-tsai) [inline ChatQnA/README.md:19]:
```suggestion
        RER([Rerank 'TEI'<br>8000])
```

File: ChatQnA/README.md [hunk 5]
```diff
@@ -10,7 +10,69 @@ ChatQnA architecture shows below:
...
+        LLM([OPEA LLM 'text generate'<br>8008])
```
Comment (louie-tsai) [inline ChatQnA/README.md:20]:
```suggestion
       LLM([LLM 'text-generation TGI'<br>8008])
```

File: ChatQnA/README.md [hunk 6]
```diff
@@ -10,7 +10,69 @@ ChatQnA architecture shows below:
...
+    direction LR
+    Ingest[Ingest data] -->|a| UI
+    UI -->|b| DP
+    DP -.->|c| TEI_EM
+
+    %% Questions interaction
+    direction LR
+    a[User Input Query] -->|1| UI
+    UI -->|2| GW
+    GW ==>|3| EM
```
Comment (louie-tsai) [inline ChatQnA/README.md:59]:
```suggestion
    GW ==>|3|  ChatQnA-MegaService
```

File: ChatQnA/README.md [hunk 7]
```diff
@@ -10,7 +10,69 @@ ChatQnA architecture shows below:
...
+    EM ==>|4| RET
+    RET ==>|5| RER
+    RER ==>|6| LLM
+    LLM ==>|7| GW
```
Comment (louie-tsai) [inline ChatQnA/README.md:63]:
```suggestion
    
```

File: ChatQnA/README.md [hunk 8]
```diff
@@ -10,7 +10,69 @@ ChatQnA architecture shows below:
...
+
+    %% Embedding service flow
+    direction TB
+    EM -.->|3'| TEI_EM
```
Comment (louie-tsai) [inline ChatQnA/README.md:67]:
```suggestion
    EM -.->|3'| TEI_EM
    RET -.->|4'| TEI_EM
```

File: ChatQnA/README.md [hunk 9]
```diff
@@ -10,7 +10,69 @@ ChatQnA architecture shows below:
...
+    RER -.->|5'| TEI_RER
+    LLM -.->|6'| LLM_gen
+
+    subgraph Legend
+        X([Micsrservice])
+        Y{{Server API}}
```
Comment (louie-tsai) [inline ChatQnA/README.md:73]:
```suggestion
        Y{{Service from industry peers}}
        Z[Gateway]
```

**Review by louie-tsai — COMMENTED** [review]
Suggest to change the flowchart as below diagram.
please help to review it.

```mermaid
graph LR
    subgraph ChatQnA-MegaService["ChatQnA-MegaService"]
        direction LR
        EM([Embedding 'LangChain TEI' <br>6000])
        RET([Retrieval 'LangChain Redis'<br>7000])
        RER([Rerank 'TEI'<br>8000])
        LLM([LLM 'text-generation TGI'<br>8008])
    end

    direction TB
    TEI_EM{{TEI embedding service<br>8090}}
    VDB{{Vector DB<br>8001}}
    %% Vector DB interaction
    TEI_EM -.->|d|VDB

    DP([OPEA Data Preparation<br>6007])
    LLM_gen{{TGI/vLLM/ollama Service}}

    direction TB
    RER([Rerank 'TEI'<br>8000])
    TEI_RER{{TEI Reranking service<br>8808}}

    subgraph User Interface
        direction TB
        a[User Input Query]
        Ingest[Ingest data]
        UI[UI server<br>Port: 5173]
    end

    subgraph ChatQnA GateWay
        direction LR
        GW[ChatQnA GateWay<br>Port: 8888]
    end

    %% Data Preparation flow
    %% Ingest data flow
    direction LR
    Ingest[Ingest data] -->|a| UI
    UI -->|b| DP
    DP -.->|c| TEI_EM

    %% Questions interaction
    direction LR
    a[User Input Query] -->|1| UI
    UI -->|2| GW
    GW <==>|3| ChatQnA-MegaService
    EM ==>|4| RET
    RET ==>|5| RER
    RER ==>|6| LLM


    %% Embedding service flow
    direction TB
    EM -.->|3'| TEI_EM
    RET -.->|4'| TEI_EM
    RER -.->|5'| TEI_RER
    LLM -.->|6'| LLM_gen

    subgraph Legend
        X([Micsrservice])
        Y{{Server API}}
    end
```

File: ChatQnA/README.md [hunk 10]
```diff
@@ -10,7 +10,71 @@ ChatQnA architecture shows below:
...
 
-![Flow Chart](./assets/img/chatqna_flow_chart.png)
+```mermaid
+graph LR
+    subgraph ChatQnA-MegaService["ChatQnA-MegaService"]
+        direction LR
+        EM([Embedding 'LangChain TEI' <br>6000])
+        RET([Retrieval 'LangChain Redis'<br>7000])
+        RER([Rerank 'TEI'<br>8000])
+       LLM([LLM 'text-generation TGI'<br>8008])
```
Comment (louie-tsai) [inline ChatQnA/README.md:20]:
```suggestion
       LLM([LLM 'text-generation TGI'<br>9000])
```

**Comment by ctao456** [comment]
Comment to review

File: ChatQnA/README.md [hunk 11]
```diff
@@ -10,7 +10,71 @@ ChatQnA architecture shows below:
...
+    direction LR
+    Ingest[Ingest data] -->|a| UI
+    UI -->|b| DP
+    DP -.->|c| TEI_EM
+
+    %% Questions interaction
+    direction LR
+    a[User Input Query] -->|1| UI
+    UI -->|2| GW
+    GW ==>|3|  ChatQnA-MegaService
```
Comment (alexsin368) [inline ChatQnA/README.md:59]: Remove extra space before "ChatQnA-MegaService". This is causing the chart to not show up on the README.

**Review by alexsin368 — CHANGES_REQUESTED** [review]
Fix error so that chart will display

File: ChatQnA/README.md [hunk 12]
```diff
@@ -10,7 +10,71 @@ ChatQnA architecture shows below:
...
+    end
+
+    direction TB
+    TEI_EM{{TEI embedding service<br>8090}}
+    VDB{{Vector DB<br>8001}}
+    %% Vector DB interaction
+    TEI_EM -.->|d|VDB
+
+    DP([OPEA Data Preparation<br>6007])
+    LLM_gen{{TGI/vLLM/ollama Service}}
```
Comment (ctao456) [inline ChatQnA/README.md:30]: `    LLM_gen{{TGI/vLLM/ollama Service<br>9009}}
`

File: ChatQnA/README.md [hunk 13]
```diff
@@ -10,7 +10,71 @@ ChatQnA architecture shows below:
...
+    subgraph ChatQnA-MegaService["ChatQnA-MegaService"]
+        direction LR
+        EM([Embedding 'LangChain TEI' <br>6000])
+        RET([Retrieval 'LangChain Redis'<br>7000])
+        RER([Rerank 'TEI'<br>8000])
+       LLM([LLM 'text-generation TGI'<br>9000])
+    end
+
+    direction TB
+    TEI_EM{{TEI embedding service<br>8090}}
```
Comment (ctao456) [inline ChatQnA/README.md:24]: `    TEI_EM{{TEI embedding service<br>6006}}
`

**Review by ctao456 — CHANGES_REQUESTED** [review]

**Review by letonghan — COMMENTED** [review]
Hi @xiguiw , the `Retrieval` microservice also need to access the `Vector DB` service. I think another arrow is needed.

**Comment by louie-tsai** [comment]
> Hi @xiguiw , the `Retrieval` microservice also need to access the `Vector DB` service. I think another arrow is needed.

Does Retrieval access VectorDB directly or access VectorDB via TEI Embedding service?

**Comment by louie-tsai** [comment]
updated flowchart according to wiki
```mermaid
---
config:
  flowchart:
    nodeSpacing: 100
    rankSpacing: 100
    curve: linear
  theme: base
  themeVariables:
    fontSize: 42px
---
flowchart LR
    %% Colors %%
    classDef blue fill:#ADD8E6,stroke:#ADD8E6,stroke-width:2px,fill-opacity:0.5
    classDef orange fill:#FBAA60,stroke:#ADD8E6,stroke-width:2px,fill-opacity:0.5
    classDef orchid fill:#C26DBC,stroke:#ADD8E6,stroke-width:2px,fill-opacity:0.5
    classDef invisible fill:transparent,stroke:transparent;
    style ChatQnA-MegaService stroke:#000000
    %% Subgraphs %%
    subgraph ChatQnA-MegaService["ChatQnA-MegaService"]
        direction LR
        EM([Embedding <br>6000]):::blue
        RET([Retrieval <br>7000]):::blue
        RER([Rerank <br>8000]):::blue
        LLM([LLM <br>9000]):::blue
    end
    subgraph User Interface
        direction TB
        a([User Input Query]):::orchid
        Ingest([Ingest data]):::orchid
        UI([UI server<br>Port: 5173]):::orchid
    end
    subgraph ChatQnA GateWay
        direction LR
        invisible1[ ]:::invisible
        GW([ChatQnA GateWay<br>Port: 8888]):::orange
    end
    subgraph .
        X([Micsrservice]):::blue
        Y{{Service from industry peers}}
        Z([Gateway]):::orange
        Z1([UI]):::orchid
    end



    %% Data Preparation flow
    %% Ingest data flow
    direction LR
    Ingest[Ingest data] -->|a| UI
    UI -->|b| DP
    DP -.->|c| TEI_EM

    %% Questions interaction
    direction LR
    a[User Input Query] -->|1| UI
    UI -->|2| GW
    GW <==>|3| ChatQnA-MegaService
    EM ==>|4| RET
    RET ==>|5| RER
    RER ==>|6| LLM


    %% Embedding service flow
    direction TB
    EM -.->|3'| TEI_EM
    RET -.->|4'| R_RET
    RER -.->|5'| TEI_RER
    LLM -.->|6'| LLM_gen

    direction TB
    TEI_RER{{Reranking service<br>'TEI'<br>8808}}
    TEI_EM{{Embedding service <br>'TEI LangChain'<br>6006}}
    R_RET{{Retriever service <br>'LangChain Redis'<br>7000}}
    VDB{{Vector DB<br>'Redis'<br>8001}}
    %% Vector DB interaction
    TEI_EM <-.->|d|VDB
    R_RET <-.->|d|VDB

    DP([Data Preparation<br>'LangChain Redis'<br>6007])
    LLM_gen{{LLM Service <br>'TGI'<br>9009}}


```

**Review by louie-tsai — APPROVED** [review]

**Pull request merged**